source/code/plugins/oms_common.rb

```diff
@@ -478,6 +480,141 @@ class Common
+        File.exist?(@@HostnameFilePath)
+      end
+
+      def is_dot_separated_string?(hnBuffer)
+        return true if /[^.]+\.[^.]+/ =~ hnBuffer
+        return false
+      end
+
+      def is_hostname_compliant?(hnBuffer)
+        # RFC 2181:
+        #   Size limit is 1 to 63 octets, so probably bytesize is appropriate method.
+        return false if hnBuffer.nil?
+        return false if /\./ =~ hnBuffer # Hostname by definition may not contain a dot.
+        return false if /:/ =~ hnBuffer # Hostname by definition may not contain a colon.
+        return true if 1 <= hnBuffer.bytesize && hnBuffer.bytesize <= 63
+        return false
+      end
+
+      def is_like_ipv4_string?(hnBuffer)
+        return false unless /\A[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\z/ =~ hnBuffer
+        qwa = hnBuffer.split('.')
+        return false unless qwa.length == 4
+        i = 0
+        qwa.each do |quadwordstring|
+            bi = quadwordstring.to_i
+            # This may need more detail if 255 octets are sometimes allowed, but I don't think so.
+            return false unless 0 <= bi and bi < 255
+            return false if i == 0 and bi == 0
+            i += 1
+        end
+        return false unless qwa[0].to_i != 0
+        return true
+      end
+
+      def is_like_ipv6_string?(hnBuffer)
+        return true if /\A\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}\z/ =~ hnBuffer
+        return false
+      end
+
+      def is_recognized_prefix_label?(fqdnLabel)
+        s = fqdnLabel.strip.downcase
+        return true if s == 'www'
+        return false
+      end
+
+      def look_for_sockety_host_address
+        hnb = nil
+
+        begin
+          socket_hostname = Socket.gethostname
+          hnb = socket_hostname.strip
+        rescue => error
+          OMS::Log.error_once("Unable to get the Host Name using socket facility: #{error}")
+          return
+        end
+
+        if is_dot_separated_string?(hnb) then
+          if is_like_ipv4_string?(hnb) then
+            # If Socket.gethostname returns an IPv4 address,
+            # use the entire string.
+            @@Hostname = hnb
+          else
+            # Here we presume we got a fully qualified domain name.
+            labelarray = hnb.split(".")
+            if is_recognized_prefix_label?(labelarray[0]) then
+              # Presume first label is a catagory aspect like www, and use the second.
+              @@Hostname = clean_hostname_string(labelarray[1])
+            else
+              # Presume first is the one.
+              @@Hostname = clean_hostname_string(labelarray.first)
+            end
+          end
+        else
+          # Here we presume we got a usable hostname,
+          # though it could also be an IPV6, or other things,
+          # and more attention may be needed here later.
+          @@Hostname = clean_hostname_string(hnb)
+        end
+      end
+
+      def look_in_Designated_HostnameFile
```

Comment:
Same casing concern here: `look_in_designated_hostname_file`

Reply:
Sure.  Out of style.  Leftovers from previous shop habits.